### PR TITLE
add some doctest to test

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: doc/src/stdlib/test.md [hunk 1]
```diff
@@ -35,28 +41,25 @@ foo (generic function with 1 method)
 
 If the condition is true, a `Pass` is returned:
 
-```julia-repl
+```jldoctest testfoo
 julia> @test foo("bar") == 9
 Test Passed
-  Expression: foo("bar") == 9
-   Evaluated: 9 == 9
+
```
Comment (tkelman) [inline doc/src/stdlib/test.md]: are all the extra blank lines needed for this to pass?

Comment (KristofferC) [inline doc/src/stdlib/test.md]: They exist in the output.

Comment (tkelman) [inline doc/src/stdlib/test.md]: is that a bug?

Comment (KristofferC) [inline doc/src/stdlib/test.md]: Not sure if you could call it a bug but it is not pretty at least.

**Comment by kshyatt** [comment]
Can we merge this?

File: doc/src/stdlib/test.md (hunk 1 above)
Comment (tkelman) [inline doc/src/stdlib/test.md]: looks wrong, anyone know what's causing it?

**Comment by KristofferC** [comment]
Probably.

File: doc/src/stdlib/test.md (hunk 1 above)
Comment (fredrikekre) [inline doc/src/stdlib/test.md]:
```diff
diff --git a/base/test.jl b/base/test.jl
index c841552..25ab49f 100644
--- a/base/test.jl
+++ b/base/test.jl
@@ -69,9 +69,9 @@ struct Pass <: Result
     value
 end
 function Base.show(io::IO, t::Pass)
-    print_with_color(:green, io, "Test Passed\n"; bold = true)
+    print_with_color(:green, io, "Test Passed"; bold = true)
     if !(t.orig_expr === nothing)
-        print(io, "  Expression: ", t.orig_expr)
+        print(io, "\n  Expression: ", t.orig_expr)
     end
     if t.test_type == :test_throws
         # The correct type of exception was thrown
```
?

Comment (KristofferC) [inline doc/src/stdlib/test.md]: Yeah, makes sense.

**Comment by KristofferC** [comment]
Pushed @fredrikekre commit and updated doctests. Should be good to go.

File: doc/src/stdlib/test.md [hunk 2]
```diff
@@ -68,23 +68,25 @@ julia> @test foo(:cat) == 1
 Error During Test
   Test threw an exception of type MethodError
   Expression: foo(:cat) == 1
-  MethodError: `length` has no method matching length(::Symbol)
```
Comment (tkelman) [inline doc/src/stdlib/test.md]: should this block go in the same jldoctest set?

File: doc/src/stdlib/test.md [hunk 3]
```diff
@@ -68,23 +68,25 @@ julia> @test foo(:cat) == 1
...
 If we expect that evaluating an expression *should* throw an exception, then we can use `@test_throws()`
 to check that this occurs:
 
-```julia-repl
+```jldoctest testfoo
 julia> @test_throws MethodError foo(:cat)
 Test Passed
-  Expression: foo(:cat)
-   Evaluated: MethodError
+
```
Comment (tkelman) [inline doc/src/stdlib/test.md]: should this blank line still be here?

File: doc/src/stdlib/test.md [hunk 4]
```diff
@@ -169,15 +170,15 @@ As calculations on floating-point values can be imprecise, you can perform appro
...
 
 julia> @test 1 ≈ 0.999999
-ERROR: test failed: 1 isapprox 0.999999
- in expression: 1 ≈ 0.999999
- in error at error.jl:21
- in default_handler at test.jl:30
- in do_test at test.jl:53
+Test Failed
+  Expression: 1 ≈ 0.999999
+   Evaluated: 1 isapprox 0.999999
```
Comment (tkelman) [inline doc/src/stdlib/test.md]: likely needs updating for #22296

Comment (KristofferC) [inline doc/src/stdlib/test.md]: Thanks, I messed up when running the doctests (they didnt actually get run) so I will rebase master and make sure everything passes.

**Comment by KristofferC** [comment]
The parse doctests fail because Documenter is not up to date.
```
> Code block:

```jldoctest
julia> ex = macroexpand( :(@sayhello("human")) )
:((println)("Hello, ", "human"))

julia> typeof(ex)
Expr
```

> Subexpression:

    ex = macroexpand( :(@sayhello("human")) )

> Output Diff:

WARNING: macroexpand(x) is deprecated, use `macroexpand(module, x)` instead.
Stacktrace:
 [1] depwarn(::String, ::Symbol) at ./deprecated.jl:70
 [2] macroexpand(::Any) at ./deprecated.jl:1376
 [3] disable_color(::Documenter.DocChecks.##8#10{Module}) at /home/kristoffer/julia/doc/deps/v0.7/Documenter/src/DocChecks.jl:558
 [4] withoutput(::Documenter.DocChecks.##7#9{Module}) at /home/kristoffer/julia/doc/deps/v0.7/Documenter/src/Utilities/Utilities.jl:587
 [5] eval_repl(::String, ::Module, ::Dict{Symbol,Any}, ::Documenter.Documents.Document, ::Documenter.Documents.Page) at /home/kristoffer/julia/doc/deps/v0.7/Documenter/src/DocChecks.jl:200
 [3] disable_color(::Documenter.DocChecks.##8#10{Module}) at /home/kristoffer/julia/doc/deps/v0.7/Documenter/src/DocChecks.jl:558
 [4] withoutput(::Documenter.DocChecks.##7#9{Module}) at /home/kristoffer/julia/doc/deps/v0.7/Documenter/src/Utilities/Utilities.jl:587
 [5] eval_repl(::String, ::Module, ::Dict{Symbol,Any}, ::Documenter.Documents.Document, ::Documenter.Documents.Page) at /home/kristoffer/julia/doc/deps/v0.7/Documenter/src/DocChecks.jl:200
```

File: doc/src/stdlib/test.md (hunk 4 above)
Comment (omus) [inline doc/src/stdlib/test.md]: Actually this was corrected as part of #22315

**Comment by KristofferC** [comment]
Doctests for this file now passes locally.

File: doc/src/stdlib/test.md (hunk 2 above)
Comment (KristofferC) [inline doc/src/stdlib/test.md]: No because the stacktrace is very weird from how Documenter evaluates things in its own module.

**Review by Sacha0 — APPROVED** [review]
lgtm! :)

**Pull request merged**